examples/squeezer_bin/squeezer_bin/imp.rs

```diff
@@ -0,0 +1,178 @@
+    type Type = super::SqueezerBin;
+
+    fn new() -> Self {
+        Self {
+            child: RefCell::new(None),
+            keep_aspect_ratio: Cell::new(false),
+        }
+    }
+}
+
+impl ObjectImpl for SqueezerBin {
+    fn properties() -> &'static [glib::ParamSpec] {
+        static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+            vec![
+                glib::ParamSpecObject::new(
+                    "child",
+                    "Child",
+                    "The child of the widget",
+                    gtk::Widget::static_type(),
+                    glib::ParamFlags::READWRITE | glib::ParamFlags::EXPLICIT_NOTIFY,
+                ),
+                glib::ParamSpecBoolean::new(
+                    "keep-aspect-ratio",
+                    "Keep aspect ratio",
+                    "If true it keeps the aspect ratio of the widget",
+                    false,
+                    glib::ParamFlags::READWRITE | glib::ParamFlags::EXPLICIT_NOTIFY,
+                ),
+            ]
+        });
+
+        PROPERTIES.as_ref()
+    }
+
+    fn property(&self, obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+        match pspec.name() {
+            "child" => self.child(obj).to_value(),
+            "keep-aspect-ratio" => self.keep_aspect_ratio(obj).to_value(),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn set_property(&self, obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+        match pspec.name() {
+            "child" => {
+                self.set_child(obj, value.get::<gtk::Widget>().ok().as_ref());
+            }
+            "keep-aspect-ratio" => {
+                self.set_keep_aspect_ratio(obj, value.get().unwrap());
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn constructed(&self, obj: &Self::Type) {
+        obj.set_halign(gtk::Align::Fill);
+        obj.set_valign(gtk::Align::Fill);
+        obj.set_hexpand(true);
+        obj.set_vexpand(true);
+
+        self.parent_constructed(obj);
+    }
+
+    fn dispose(&self, obj: &Self::Type) {
+        if let Some(child) = obj.child() {
+            child.unparent();
+        }
+    }
+}
+
+impl WidgetImpl for SqueezerBin {
+    fn size_allocate(&self, widget: &Self::Type, width: i32, height: i32, baseline: i32) {
+        let ((_, horizontal_size), (_, vertical_size)) = self.child_size(widget);
+
+        let (mut horizontal_zoom, mut vertical_zoom) = (
+            width as f32 / horizontal_size as f32,
+            height as f32 / vertical_size as f32,
+        );
+
+        if widget.keep_aspect_ratio() {
+            if horizontal_zoom < vertical_zoom {
+                vertical_zoom = horizontal_zoom;
+            } else {
+                horizontal_zoom = vertical_zoom;
+            }
+        }
+
+        let transform = gsk::Transform::new()
+            .scale(horizontal_zoom, vertical_zoom)
+            .unwrap();
+
+        widget.first_child().unwrap().allocate(
```

Comment:
This call and all other calls to `first_child()` could just use the `child` you've stored internally now anyway or not?